src/sentry/static/sentry/app/data/forms/projectDebugFiles.jsx

```diff
@@ -18,30 +55,74 @@ export const fields = {
   },
   symbolSources: {
     name: 'symbolSources',
-    type: 'string',
+    type: 'rich_list',
     label: t('Custom Repositories'),
-    placeholder: t('Paste JSON here.'),
-    multiline: true,
-    monospace: true,
-    autosize: true,
-    inline: false,
-    maxRows: 10,
-    saveOnBlur: false,
-    saveMessageAlertType: 'info',
-    saveMessage: t('Updates will apply to future events only.'),
+    help: t('Configures custom repositories containing debug files.'),
     formatMessageValue: false,
-    help: t(
-      'Configures custom repositories containing debug files. At the moment, only Amazon S3 buckets are supported.'
-    ),
-    validate: ({id, form}) => {
-      try {
-        if (form[id].trim()) {
-          JSON.parse(form[id]);
-        }
-      } catch (e) {
-        return [[id, e.toString().replace(/^SyntaxError: JSON.parse: /, '')]];
+    addButtonText: t('Add Repository'),
+    addDropdown: {
+      items: [
+        {
+          value: 's3',
+          label: sourceNames.s3,
+          searchKey: t('aws amazon s3 bucket'),
+        },
+        {
+          value: 'gcs',
+          label: sourceNames.gcs,
+          searchKey: t('gcs google cloud storage bucket'),
+        },
+        {
+          value: 'http',
+          label: sourceNames.http,
+          searchKey: t('http symbol server ssqp symstore symsrv'),
+        },
+      ],
+    },
+
+    getValue: sources => JSON.stringify(sources.map(unflattenKeys)),
+    setValue: raw => (JSON.parse(raw || null) || []).map(flattenKeys),
+
+    renderItem(item) {
+      if (item.name) {
+        return item.name;
+      } else {
+        return <em>{t('<Unnamed Repository>')}</em>;
       }
-      return [];
+    },
+
+    onAddItem(item, addItem) {
+      openDebugFileSourceModal({
+        sourceType: item.value,
+        onSave: addItem,
+      });
+    },
+
+    onEditItem(item, updateItem) {
+      openDebugFileSourceModal({
+        sourceConfig: item,
+        sourceType: item.type,
+        onSave: updateItem,
+      });
+    },
+
+    removeConfirm: {
+      title: t('Remove Repository?'),
+      confirmText: t('Remove Repository'),
+      message: (
+        <>
```

Comment:
Are we using this syntax? In most of the application we use `<React.Fragment>` as `<>` is un-googlable to folks who don't know what it is.

Reply:
@EvanPurkhiser or @billyvg, can you please inform on this?

My take on this is that developers touching this should know what they are doing and `React.Fragment` is just lengthy, but I'll let you decide on this.

Reply:
👎 on this syntax for the moment. Until we have a eslint rule to be consistent, lets stick with `React.Fragment`.